### PR TITLE
win/ci: Link the Envoy bin locally to prevent OOM

### LINKED ISSUE #30146: Windows CI is OOMing
```
LINK : fatal error LNK1102: out of memory
```

https://dev.azure.com/cncf/envoy/_build/results?buildId=152223&view=logs&j=b840a642-5ff3-5357-2e4b-e06e40b0cffd&t=67965174-5100-5631-9dc0-68b9f0aacb53&l=12401

### PR DESCRIPTION
**Pull request by phlax**
Fix #30146 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @akshaysngupta  need to get confirmation from CI - it might be we need to boost VM if we do this - but im pretty sure the combination will work

File: .github/workflows/_ci.yml [hunk 1]
```diff
@@ -193,6 +207,17 @@ jobs:
         echo "disk space at end of build:"
         df -h
         echo
-        du -ch "${{ runner.temp }}" | grep -E "[0-9]{2,}M|[0-9]G"
+        if [[ "${{ inputs.run-du }}" != "false" ]]; then
+            du -ch "${{ inputs.temp-dir || runner.temp }}" | grep -E "[0-9]{2,}M|[0-9]G"
+        fi
       name: "Check disk space at end"
       shell: bash
+<<<<<<< HEAD
```
Comment (akshaysngupta) [inline .github/workflows/_ci.yml:215]: remove

Comment (phlax) [reply .github/workflows/_ci.yml:215]: argh - i pushed to the wrong pr - ill cleanup

File: .github/workflows/_env.yml [hunk 2]
```diff
@@ -106,7 +106,7 @@ concurrency:
 
 jobs:
   repo:
-    if: github.repository == 'envoyproxy/envoy'
+    # if: github.repository == 'envoyproxy/envoy'
```
Comment (akshaysngupta) [inline .github/workflows/_env.yml:109]: what's the reason for this `if` condition ? are these yml used in other repos as well ?

Comment (phlax) [reply .github/workflows/_env.yml]: see comment above

**Review by akshaysngupta — APPROVED** [review]

**Comment by phlax** [comment]
closing in favour of #30538

**Pull request closed without merging**